core/modules/wdb/QueryAction.cc

```diff
@@ -164,6 +167,9 @@ QueryAction::Impl::Impl(QueryActionArg const& a)
       _poisoned(false),
       _sendChannel(a.task->sendChannel),
       _user(a.task->user) {
+    char buf[MAX_NAME_LENGTH];
+    gethostname(buf, MAX_NAME_LENGTH);
+    _hostName = buf;
```

Comment:
I added a mutex and wound up with this

std::string const& getHostname() {
    static std::mutex mx;
    static std::string cachedHostname;
    std::lock_guard<std::mutex> lg(mx);
    if (cachedHostname.empty()) {
        char buf[HOST_NAME_MAX+1];
        buf[HOST_NAME_MAX] = '\0';
        gethostname(buf, sizeof buf);
        cachedHostname = buf;
    }
    return cachedHostname;
}